### PR TITLE
quick fixes UI support

### PR DESCRIPTION
**Pull request by kasperpeulen**

### CODE REVIEW

**Comment by devoncarew** [comment]
Sorry for the delay - will take a look tomorrow.

File: lib/editing/editor_codemirror.css [hunk 1]
```diff
@@ -177,6 +177,17 @@ li.CodeMirror-hint-active {
...
 }
 
+.type-quick_fix {
+  font-family: "Inconsolata", sans-serif;
+  font-size: 12pt;
+}
+
+.type-quick_fix::before {
+  content: " ";
+  background: url('../images/wrench.svg');
```
Comment (devoncarew) [inline lib/editing/editor_codemirror.css]: Can you move this svg into the same folder as the css file? I think that'll prevent issues with the deployed build.

File: lib/editing/editor_codemirror.dart [hunk 2]
```diff
@@ -181,6 +198,18 @@ class _CodeMirrorEditor extends Editor {
 
   void execCommand(String name) => cm.execCommand(name);
 
+  void showCompletions({bool autoInvoked: false, bool onlyShowFixes: false}) {
+    if (autoInvoked)
```
Comment (devoncarew) [inline lib/editing/editor_codemirror.dart]: Can you add braces (`{`) here and below? The dart style guide requires them except for single-line if statements.

File: lib/elements/elements.css [hunk 3]
```diff
@@ -163,6 +163,11 @@
   /*box-shadow: 3px 3px 4px 0px rgba(15, 14, 14, 0.7);*/
 }
 
+#issues .issue.hasFix {
+  background: #444 url('../images/wrench.svg') right no-repeat;
```
Comment (devoncarew) [inline lib/elements/elements.css]: Hmm, or perhaps but the image here, and refer to it using a relative path from the codemirror editor's css?

File: lib/services/common.dart [hunk 4]
```diff
@@ -4,9 +4,13 @@
...
 
-final String serverURL = 'https://dart-services.appspot.com/';
-//final String serverURL = 'http://doc-test.dart-services.appspot.com/';
-//final String serverURL = 'http://127.0.0.1:8082/';
+
+// TODO: This is just for testing this PR remove when the dart-services are
+// updated and this PR is ready for merge
+
+//final String serverURL = 'https://dart-services.appspot.com/';
+final String serverURL = 'http://doc-test.dart-services.appspot.com/';
```
Comment (devoncarew) [inline lib/services/common.dart]: Let's remove this, coordinate with @lukechurch re: the server, and plan to land this PR soon.

**Comment by devoncarew** [comment]
Some small changes. Can you:

- merge this PR with master, and re-push it to here?
- coordinate with @lukechurch re: the server change and landing this PR?

**Comment by kasperpeulen** [comment]
@devoncarew merged with master now

About the server, it seems that the quick fixes are already in the server. There is one issue though, that the hasFix bool of the dart-services is not really reliable. So sometimes a wrench fix is added, while there is no fix, or the other way around.

**Comment by kasperpeulen** [comment]
fix #260

**Comment by lukechurch** [comment]
@kasperpeulen 
>There is one issue though, that the hasFix bool of the dart-services is not really reliable.

For various scalability how to use a heuristic to compute this. The heuristic is conservative, so I'm not at all surprised that it is sometimes false when it should be true.

The opposite is more surprising. Please can you give me an example?

**Comment by devoncarew** [comment]
merging...

**Comment by devoncarew** [comment]
I see this with an unused import: https://dartpad.dartlang.org/459e0887f387ad463528

**Comment by lukechurch** [comment]
What I'm looking for is cases where the wrench appears but there isn't a quick fix available.

**Comment by kasperpeulen** [comment]
@devoncarew I found some bugs with the quick fixes, but all those bug doesn't appear if I run the master of dart-services locally. So I guess this will be solved very soon, as l heard lukechurch plans to update dart-services tonight.

**Comment by devoncarew** [comment]
OK, thanks for the update. This isn't deployed - and we don't plan to rev a new public version until ~next Wednesday, so I think we're good as is.

**Pull request merged**